cads_adaptors/adaptors/cams_solar_rad/functions.py

```diff
@@ -0,0 +1,483 @@
+import hashlib
+import logging
+import re
+import time
+import traceback
+
+import jinja2
+
+
+class BadRequest(Exception):
+    pass
+
+
+class NoData(Exception):
```

Comment:
as previous comment

Reply:
The BadRequest and NoData exceptions are caught before getting outside the adaptor.